internal/controllers/machineset/machineset_controller.go

```diff
@@ -355,15 +373,17 @@ func (r *Reconciler) reconcile(ctx context.Context, cluster *clusterv1.Cluster,
 	if machineSet.Spec.MinReadySeconds > 0 &&
 		machineSet.Status.ReadyReplicas == replicas &&
 		machineSet.Status.AvailableReplicas != replicas {
-		return ctrl.Result{RequeueAfter: time.Duration(machineSet.Spec.MinReadySeconds) * time.Second}, nil
+		minReadyResult := ctrl.Result{RequeueAfter: time.Duration(machineSet.Spec.MinReadySeconds) * time.Second}
+		result = util.LowestNonZeroResult(result, minReadyResult)
+		return result, nil
 	}
 
 	// Quickly reconcile until the nodes become Ready.
 	if machineSet.Status.ReadyReplicas != replicas {
-		return ctrl.Result{RequeueAfter: 15 * time.Second}, nil
+		result = util.LowestNonZeroResult(result, ctrl.Result{RequeueAfter: 15 * time.Second})
```

Comment:
```suggestion
		result = util.LowestNonZeroResult(result, ctrl.Result{RequeueAfter: 15 * time.Second})
		return result, nil
```
nit: for consistency and to be more explicit